### PR TITLE
Closes #2683: Update Campus Safety link in Global footer

### LINKED ISSUE #2683: Update 'Campus Safety' link in global footer to point to Safety.arizona.edu
Hello, with the new Safety.arizona.edu subdomain now being live, we'll be unpublishing and redirecting the 'Campus Safety, Security & Emergency Info' page on Arizona.edu (https://www.arizona.edu/campus-safety) to this new site. Can the CWS team please update the 'Campus Safety' link in the global footer (image attached showing link placement)? The Web Experience team in Marketing and Communications can create the redirect on Arizona.edu in the meantime. 
Thank you!
- Catherine Gustafson
Monosnap Campus Safety, Security   Emergency Info | The University of Arizona, Tucson, Arizona 2023-08-21 14-14-13

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
- Updates Campus Safety link in data used to build Global Footer menus on new module installs
- Adds DB update to Global Footer module for updating Campus Safety links on sites that already installed the module.

## Related issues

Closes #2683

### On a new site (e.g. [ProboCI](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2684.probo.build/))
- Install Global Footer module
- Verify that Campus Safety Link is correct in Global footer

### On existing site
- Apply PR (e.g. as composer patch)
- Run DB updates
- Verify that Campus Safety Link is correct in Global footer

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [x] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_global_footer/az_global_footer.install [hunk 1]
```diff
@@ -44,3 +44,37 @@ function az_global_footer_update_last_removed() {
...
   return 920501;
 }
+
+/**
+ * Update Campus Safety link(s)
+ *
+ * Campus Saftey link(s) in the global footer will be updated with
+ * new safety.arizona.edu domain.
+ */
+function az_global_footer_update_920701() {
```
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:54]:
```suggestion
function az_global_footer_update_1020701() {
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 2]
```diff
@@ -44,3 +44,37 @@ function az_global_footer_update_last_removed() {
...
+    ->condition('link__uri', 'https://www.arizona.edu/campus-safety')
+    ->execute();
+
+  foreach ($menu_link_content_ids as $id) {
+    /**
+     * @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content
+     */
+    $menu_link_content = MenuLinkContent::load($id);
+    if ($menu_link_content !== NULL) {
+      $menu_link_content->title = 'International Engagement';
```
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:69]:
```suggestion
      $menu_link_content->title = 'Safety';
```

Comment (trackleft) [reply modules/custom/az_global_footer/az_global_footer.install:69]: But we probably don't even need to update the title.

File: modules/custom/az_global_footer/az_global_footer.install [hunk 3]
```diff
@@ -44,3 +44,37 @@ function az_global_footer_update_last_removed() {
...
+ *
+ * Campus Saftey link(s) in the global footer will be updated with
+ * new safety.arizona.edu domain.
+ */
+function az_global_footer_update_920701() {
+  $updated_count = 0;
+  $menu_link_content_ids = \Drupal::entityQuery('menu_link_content')
+    ->accessCheck(FALSE)
+    ->condition('menu_name', 'az-footer-main')
+    ->condition('title', 'Campus Saftey')
```
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:59]:
```suggestion
    ->condition('title', 'Campus Safety')
```

File: modules/custom/az_global_footer/az_global_footer.install (hunk 2 above)
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:69]:
```suggestion
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 4]
```diff
@@ -44,3 +44,37 @@ function az_global_footer_update_last_removed() {
   // Remove updates added before 2.6.0.
   return 920501;
 }
+
+/**
+ * Update Campus Safety link(s)
+ *
+ * Campus Saftey link(s) in the global footer will be updated with
```
Comment (bberndt-uaz) [inline modules/custom/az_global_footer/az_global_footer.install:51]:
```suggestion
 * Campus Safety link(s) in the global footer will be updated with
```

**Review by trackleft — APPROVED** [review]

File: modules/custom/az_global_footer/az_global_footer.install [hunk 5]
```diff
@@ -44,3 +44,36 @@ function az_global_footer_update_last_removed() {
...
+ *
+ * Campus Safety link(s) in the global footer will be updated with
+ * new safety.arizona.edu domain.
+ */
+function az_global_footer_update_1020701() {
+  $updated_count = 0;
+  $menu_link_content_ids = \Drupal::entityQuery('menu_link_content')
+    ->accessCheck(FALSE)
+    ->condition('menu_name', 'az-footer-main')
+    ->condition('title', 'Campus Safety')
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:59]: We could probably remove this title condition.

**Review by bberndt-uaz — APPROVED** [review]
Installed the Global Footer on the Probo site and confirmed that the Campus Safety link is correct.

**Review by camikazegreen — APPROVED** [review]
Looks like it meets all the requirements and didn't cause any regressions.

**Pull request merged**